### PR TITLE
Clear Finished Tasks 

### PR DESCRIPTION
**Pull request by nightism**

### CODE REVIEW

**Comment by nightism** [comment]
@pyokagan rebased already

**Comment by nightism** [comment]
Emmm.. I think something wrong when rebasing the last two commit....
But it's fine since it does not affect the funtionality

**Comment by nightism** [comment]
@pyokagan will this be merged? I think for now there is no major issue regarding to this function.

File: src/main/java/seedu/address/model/TaskBook.java [hunk 1]
```diff
@@ -93,6 +102,14 @@ public DeadlineTask removeDeadlineTask(int index) {
         return deadlineTasks.remove(index);
     }
 
+    /**
+     * Remove all the deadline tasks that satisfy the given predicate.
+     * @param predicate The filter used to judge whether a deadline task needs to be deleted or not.
+     */
+    public boolean removeDeadlineTasks(Predicate<? super DeadlineTask> predicate) {
+        return deadlineTasks.removeIf(predicate);
+    }
```
Comment (pyokagan) [inline src/main/java/seedu/address/model/TaskBook.java:111]: Hmph. You didn't address my comments in the last round.

File: src/main/java/seedu/address/model/WorkingTaskBook.java [hunk 2]
```diff
@@ -119,6 +119,27 @@ public FloatingTask removeFloatingTask(int workingIndex) throws IllegalValueExce
         return removedFloating;
     }
 
+    /**
+     * Remove all the floating tasks that satisfy the given predicate.
+     * @param predicate The filter used to judge whether a floating task needs to be deleted or not.
+     * @throws IllegalValueException Will be throw during iteration if internal error happens.
```
Comment (pyokagan) [inline src/main/java/seedu/address/model/WorkingTaskBook.java:125]: I don't see any `throws` statement.

**Comment by pyokagan** [comment]
The main problem is that the `clear fin` command is that it is now inconsistent with the `clear` command. `clear fin` only works with the current filtered tasks, while `clear` works on the entire task book. Going to be annoying to explain that in the user guide...

File: src/main/java/seedu/address/model/TaskBook.java (hunk 1 above)
Comment (nightism) [reply src/main/java/seedu/address/model/TaskBook.java:111]: oh,, sorry.. forgot to delete

**Pull request merged**